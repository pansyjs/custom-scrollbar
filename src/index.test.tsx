import React from 'react';
import { CustomScrollbar } from './';
import { render, waitFor } from '@testing-library/react';

test('renders without crashing', () => {
  const { container } = render(
    <div>
      <CustomScrollbar>
        {[...Array(5)].map((_, i) => (
          <p key={i}>Some content</p>
        ))}
      </CustomScrollbar>
    </div>
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('renders with options', () => {
  const { container } = render(
    <CustomScrollbar forceVisible="y">
      {[...Array(5)].map((x, i) => (
        <p key={i}>Some content</p>
      ))}
    </CustomScrollbar>
  );
  expect(container.firstChild).toMatchSnapshot();

  expect(
    container.querySelector('.custom-scrollbar-track.custom-scrollbar-vertical')
  ).toBeVisible();
});
