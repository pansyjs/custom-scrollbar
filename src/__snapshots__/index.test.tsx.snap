// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with options 1`] = `
<div
  class="custom-scrollbar-scrollable-y"
  data-simplebar="init"
>
  <div
    class="custom-scrollbar-wrapper"
  >
    <div
      class="custom-scrollbar-height-auto-observer-wrapper"
    >
      <div
        class="custom-scrollbar-height-auto-observer"
      />
    </div>
    <div
      class="custom-scrollbar-mask"
    >
      <div
        class="custom-scrollbar-offset"
        style="bottom: 0px;"
      >
        <div
          aria-label="scrollable content"
          class="custom-scrollbar-content-wrapper"
          role="region"
          style="height: auto; overflow-x: hidden; overflow-y: scroll;"
          tabindex="0"
        >
          <div
            class="custom-scrollbar-content"
          >
            <p>
              Some content
            </p>
            <p>
              Some content
            </p>
            <p>
              Some content
            </p>
            <p>
              Some content
            </p>
            <p>
              Some content
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="custom-scrollbar-placeholder"
      style="width: 0px; height: 0px;"
    />
  </div>
  <div
    class="custom-scrollbar-track custom-scrollbar-horizontal"
    style="visibility: hidden;"
  >
    <div
      class="custom-scrollbar-scrollbar"
      style="width: -8px; display: none;"
    />
  </div>
  <div
    class="custom-scrollbar-track custom-scrollbar-vertical"
    style="visibility: visible;"
  >
    <div
      class="custom-scrollbar-scrollbar"
      style="height: -8px; display: none;"
    />
  </div>
</div>
`;

exports[`renders without crashing 1`] = `
<div>
  <div
    class=""
    data-simplebar="init"
  >
    <div
      class="custom-scrollbar-wrapper"
    >
      <div
        class="custom-scrollbar-height-auto-observer-wrapper"
      >
        <div
          class="custom-scrollbar-height-auto-observer"
        />
      </div>
      <div
        class="custom-scrollbar-mask"
      >
        <div
          class="custom-scrollbar-offset"
          style="right: 0px; bottom: 0px;"
        >
          <div
            aria-label="scrollable content"
            class="custom-scrollbar-content-wrapper"
            role="region"
            style="height: auto; overflow-x: hidden; overflow-y: hidden;"
            tabindex="0"
          >
            <div
              class="custom-scrollbar-content"
            >
              <p>
                Some content
              </p>
              <p>
                Some content
              </p>
              <p>
                Some content
              </p>
              <p>
                Some content
              </p>
              <p>
                Some content
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="custom-scrollbar-placeholder"
        style="width: 0px; height: 0px;"
      />
    </div>
    <div
      class="custom-scrollbar-track custom-scrollbar-horizontal"
      style="visibility: hidden;"
    >
      <div
        class="custom-scrollbar-scrollbar"
        style="width: -8px; display: none;"
      />
    </div>
    <div
      class="custom-scrollbar-track custom-scrollbar-vertical"
      style="visibility: hidden;"
    >
      <div
        class="custom-scrollbar-scrollbar"
        style="height: -8px; display: none;"
      />
    </div>
  </div>
</div>
`;
